anymail/webhooks/mailpace.py

```diff
@@ -0,0 +1,145 @@
+import base64
+import binascii
+import json
+
+from django.utils import timezone
+from django.utils.dateparse import parse_datetime
+
+from anymail.exceptions import (
+    AnymailConfigurationError,
+    AnymailImproperlyInstalled,
+    AnymailWebhookValidationFailure,
+    _LazyError,
+)
+from anymail.utils import get_anymail_setting
+
+try:
+    from nacl.exceptions import CryptoError, ValueError
+    from nacl.signing import VerifyKey
+except ImportError:
+    # This will be raised if verification is attempted (and pynacl wasn't found)
+    VerifyKey = _LazyError(
+        AnymailImproperlyInstalled(missing_package="pynacl", install_extra="mailpace")
+    )
+
+
+from ..inbound import AnymailInboundMessage
+from ..signals import (
+    AnymailInboundEvent,
+    AnymailTrackingEvent,
+    EventType,
+    RejectReason,
+    inbound,
+    tracking,
+)
+from .base import AnymailBaseWebhookView
+
+
+class MailPaceBaseWebhookView(AnymailBaseWebhookView):
+    """Base view class for MailPace webhooks"""
+
+    esp_name = "MailPace"
+
+    def parse_events(self, request):
+        esp_event = json.loads(request.body.decode("utf-8"))
+        return [self.esp_to_anymail_event(esp_event)]
+
+
+class MailPaceTrackingWebhookView(MailPaceBaseWebhookView):
+    """Handler for MailPace delivery webhooks"""
+
+    webhook_key = None
+
+    def __init__(self, **kwargs):
+        try:
+            self.webhook_key = get_anymail_setting(
+                "webhook_key", esp_name=self.esp_name, kwargs=kwargs, allow_bare=True
+            )
+            self.warn_if_no_basic_auth = False
+        except AnymailConfigurationError:
```

Comment:
You can pass `default=None` to `get_anymail_setting` to avoid the exception, and then just check if the returned value is None.